### PR TITLE
magma: 2.0.2 -> 2.5.0

### PR DESCRIPTION
**Pull request by tbenst**
###### Motivation for this change
Without magma, many linalg functions in pytorch are broken on GPU. I also had to update magma to support newer versions of CUDA.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by tbenst** [comment]
@ianwookim let me know if you have any input!

**Comment by stites** [comment]
Heads-up, MAGMA also allows for building with MKL support and I might try to shoe-horn this build optimization into #65041 or layer on an additional PR here.

**Comment by tbenst** [comment]
Now that https://github.com/NixOS/nixpkgs/pull/65041#issuecomment-526338435 is ready, this is blocking. Would someone be able to review? Thanks!

**Comment by stites** [comment]
It's worth pointing out that I have been using this code and running the full pytorch test suite with it.

File: pkgs/development/libraries/science/math/magma/default.nix
Comment (FRidh) [inline pkgs/development/libraries/science/math/magma/default.nix]: Why does this copy a part of the stdenv?

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by wavewave** [comment]
@tbenst thank you for upgrading this! :+1:

File: pkgs/development/libraries/science/math/magma/default.nix
Comment (tbenst) [reply pkgs/development/libraries/science/math/magma/default.nix]: We need to run “make magma” and then run make for sparse. Let me know if I missed a more elegant way to do this? I don’t believe the makeflags options are sufficient as we need to run two separate make commands

Comment (FRidh) [reply pkgs/development/libraries/science/math/magma/default.nix]: You can pass `buildFlags = [ "magma" "magma_sparse" ];`

**Comment by tbenst** [comment]
Thanks @FRidh your suggestion worked. @wavewave should I update your handle from ianwookin to wavewave?

**Review by wavewave — APPROVED** [review]
I checked this works.  btw, please take the maintainership since I am not very familar with recent updates of magma. :-) 
(and it is more desirable to be listed as `ianwookim` as listed in `/maintainers/maintainer-list.nix`.)
Thanks again! @tbenst

**Comment by FRidh** [comment]
undefined variable 'wavewave' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/eval-2-shlevy.ewr1.nix.ci/pkgs/development/libraries/science/math/magma/default.nix:46:39

**Comment by tbenst** [comment]
@wavewave @FRidh done!

**Comment by tbenst** [comment]
Just reviewed https://github.com/NixOS/nixpkgs/pull/65041 and realized that @stites's excellent work building on-top of this pull request for adding MKL might best be added here. Let me know if you want to be added as a maintainer?

**Comment by stites** [comment]
I'm probably not a good maintainer for magma -- it's a slightly magical build system which I haven't fully figured out. I'd be happy to be added as a maintainer as a second pair of eyes, however.

**Comment by tbenst** [comment]
Ok, just wanted to make sure that I acknowledged your contributions to this pull request :). In that case I think this is good to go!

**Pull request merged**
